src/views/pages/Inbox/index.js

```diff
@@ -1,24 +1,32 @@
 import React, {useEffect, useState} from 'react';
-import MessageService from '../../../services/Message.service';
+import MessageAPIService from '../../../services/MessageAPI';
 import ListView from '../../components/ListView';
+import PropTypes from 'prop-types';
 import PageContainer from '../../components/PageContainer';
 import PageTitle from '../../components/PageTitle';
 import InboxSubNav from '../../components/InboxSubNav';
 
-const message = new MessageService();
+const message = new MessageAPIService();
 
-function Inbox() {
+function Inbox(props) {
   const [messageList, setMessageList] = useState([]);
 
   useEffect(() => {
-    const sub = message.messages$.subscribe((d) => {
+    const sub = message.received_messages$.subscribe((d) => {
       setMessageList(d);
     });
 
+    message.checkMessages(props.user_identity);
+
+    let id = setInterval(() => {
+      message.checkMessages(props.user_identity);
+    }, 5000);
+
     return () => {
       sub.remove();
+      clearInterval(id);
     };
-  }, []);
+  }, [props.user_identity]);
```

Comment:
This makes sense. Good work.

Only thing you might run into some trouble with is connecting the messages that get pulled down using `checkMessages` back to the `received_messages` -- the `received_messages` aren't being filtered by a `user_identity`, but only end up showing you messages from that identity since this is the only active component calling `checkMessages`. 

This is likely going to be fine for the time being, so its not something worry about at this point. Just thought it would be good to point out.